sys/net/link_layer/ble_ll/ble_ll.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (C) 2015 James Hollister
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     ble_ll
+ * @file
+ * @brief       Implementation of the bluetooth low energy link layer
+ *
+ * @author      James Hollister <jhollisterjr@gmail.com>
+ * @}
+ */
+
+#include <errno.h>
+
+#include "kernel.h"
+#include "msg.h"
+#include "thread.h"
+#include "net/ng_nomac.h"
+#include "net/ng_netbase.h"
+#include "net/ble_ll.h"
+
+#define ENABLE_DEBUG    (1)
+#define ENABLE_BLE_PKT_DEBUG	(0)
+#include "debug.h"
+
+/**
+ * @brief   Possible BLE link layer states
+ */
+typedef enum {
+    LL_STANDBY,		    /**< link layer is in standby mode */
+    LL_ADVERTISING,         /**< link layer is in advertising mode */
+    LL_CONNECTION,          /**< link layer is in connection mode */
+    LL_INITIATING,          /**< link layer is in initiating mode */
+    LL_SCANNING,            /**< link layer is in scanning mode */
+} ll_state_t;
+
+/**
+ * @brief   Current state of the BLE Link Layer
+ */
+static volatile ll_state_t _ll_state = LL_ADVERTISING;
+
+/**
+ * @brief   BLE Advertising address
+ */
+static uint8_t _adv_addr[BLE_ADDR_LEN + 1];
```

Comment:
Could you shortly explain why you need to initialize it `BLE_ADDR_LEN` __`+ 1`__ long?